docs/DeveloperGuide.md

```diff
@@ -347,14 +408,206 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+4.  ContactCS marks these contacts for quick access
+
+**Extensions**
+
+* 2a. The list is empty.
+
+    Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. ContactCS shows an error message.
+
+      Use case resumes at step 2.
+
+* 4a. The given format is invalid.
+
+    * 4a1. ContactCS shows an error message prompting the user to provide the correct format
 
       Use case resumes at step 2.
 
+**Use case: Access frequently used contacts**
+
+**MSS**
+
+1.  User requests to access frequently used contacts
+2.  ContactCS shows a list of persons
+
+**Extensions**
+
+* 2a. The list is empty.
+
+    Use case ends.
+
+* 3a. The given format is invalid.
+
+    * 3a1. ContactCS shows an error message prompting the user to provide the correct format
+
+      Use case ends.
+
+**Use case: View usage instructions**
+
+**MSS**
+
+1.  User requests to view usage instructions for the app
+2.  ContactCS displays the usage instructions and commands available for the user
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The given format is invalid.
+
+    * 2a1. ContactCS shows an error message prompting the user to provide the correct format
+
+        Use case ends.
+
+**Use case: View app with sample data**
+
+**MSS**
+
+1.  User requests to view app with sample data
+2.  ContactCS populates the interface with sample data for demonstration purposes
+3.  User interacts with the app to see how it functions with the sample data
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The given format is invalid.
+
+    * 2a1. ContactCS shows an error message prompting the user to provide the correct format
+
+      Use case ends.
+
+**Use case: Redo/Undo actions**
+
+**MSS**
+
+1.  User performs an action in ContactCS
+2.  User requests to undo the last action
+3.  ContactCS reverts to the state before the last action
+4.  If the user wishes to redo the action, they request a redo
+5.  ContactCS reapplies the last undone action
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. No action has been performed yet.
+    * 2a1. ContactCS shows an error message indicating that there is no action to undo
+
+       Use case ends.
+
+* 3a. User tries to redo an action without having undone one first.
+    * 3a1. ContactCS shows an error message indicating that there is no action to redo
+
+        Use case ends.
+
+* 4a. The given format is invalid.
+
+    * 4a1. ContactCS shows an error message prompting the user to provide the correct format
+
+      Use case ends.
+
+**Use case: Open Outlook for email communication**
+
+**MSS**
+
+1.  User requests to open the email client from ContactCS
+2.  ContactCS opens the Outlook web application in the default browser, pre-filled with the selected contact’s email address
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The user does not have internet access.
+    * 2a1. ContactCS shows an error message indicating that there is no internet connection
+
+        Use case ends.
```

Comment:
Thanks for your feedback about the auto-open feature for Outlook! I see where you’re coming from, especially with the tP constraints saying it’s unlikely to have. However, it’s mentioned in the user story in the DG. Should we keep it in the user story but remove it from use cases or remove it completely from DG? @MAOXIONGKAI

Reply:
Personally I prefer keeping the user story but deleting such use case, because user story is optional to implement but use case is really specifying what our system can do and it will serve as a more detailed requirement specification which we would need when developing the actual product ourselves.

PS: Sorry if there's confusion but I was trying to say that our user story for opening outlook is classified as unlikely-to-have, not from the tP-constraints, sorry for the confusion

Reply:
i think i agree, let's avoid interfacing with external stuff if possible

we could still make sending emails somewhat seamless if we have like a "copy to clipboard" button instead!